docs/python/guides/testing.md

```diff
@@ -5,4 +5,69 @@ sidenav: side-nav-python.html
 type: markdown
 ---
 
-## Testing
\ No newline at end of file
+# Testing
+
+## Introduction
+
+Abseil Python's Testing library is similar to Python's standard `unittest`
+module (sometimes referred to as PyUnit) but offers some additional useful
+features on top of the standard library, such as interfacing with Abseil Flags.
+
+When using Abseil Python Test, take the following steps in your unit tests:
+
+* import the `absltest` module
+* import the `flags` module, which gives you access to the variables
+`FLAGS.test_srcdir` and `FLAGS.test_tmpfir`.
+* call `absltest.main()` instead of `unittest.main()`
+
+## Unit Tests Basics
+
+Within a unit test class, any method name starting with 'test' will be run
+automatically as part of the unit test. Test names should describe the
+particular case being tested.
+
+### What is Unittest/Absltest Doing?
+
+The pattern is simple and common to most xUnit frameworks. Basically:
+
+*   `main()` scans the current file for classes derived from `TestCase`
+*   `main()` then scans each class for methods prefixed by `"test"` (e.g.
+    `testInitialize(self):`)
+*   then for each class and each test method it:
```

Comment:
then for each class and each test method it -> For each class and each test method, the framework: